DVSRegister/Middleware/BasicAuthMiddleware.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System.Net.Http.Headers;
+using System.Text;
+using Microsoft.Extensions.Options;
+
+namespace DVSRegister.Middleware
+{
+    // You may need to install the Microsoft.AspNetCore.Http.Abstractions package into your project
+    public class BasicAuthMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly BasicAuthMiddlewareConfiguration configuration;
+
+        public BasicAuthMiddleware(RequestDelegate next, IOptions<BasicAuthMiddlewareConfiguration> options)
+        {
+            _next = next;
+            configuration = options.Value;
+        }
+
+        public async Task Invoke(HttpContext httpContext)
+        {
+
+            if (IsAuthorised(httpContext))
+            {
+                await _next.Invoke(httpContext);
+            }
+            else
+            {
+                SendUnauthorisedResponse(httpContext);
+            }
+        }
+        private bool IsAuthorised(HttpContext httpContext)
+        {
+            try
+            {
+                var authHeader = AuthenticationHeaderValue.Parse(httpContext.Request.Headers["Authorization"]);
+                var credentialBytes = Convert.FromBase64String(authHeader.Parameter);
+                var credentials = Encoding.UTF8.GetString(credentialBytes).Split(new[] { ':' }, 2);
+                var username = credentials[0];
+                var password = credentials[1];
+
+                var returnValue = configuration.Username == username
+                       && configuration.Password == password;
+                Console.WriteLine("Configuration Username and Password: "+configuration.Username+"-"+configuration.Password);
+                Console.WriteLine("Given Username and Password: " + username + "-" + password);
+                Console.WriteLine("Computed Return Value: " +returnValue);
+
+                return returnValue;
+            }
+            catch
+            {
+                // Default to denying access if anything goes wrong
```

Comment:
Could we log the exception, in case of any exception?

Reply:
Added log